flex-templates/java/regional_dlp_transform/src/main/java/com/google/cloud/blueprints/datawarehouse/Util.java

```diff
@@ -0,0 +1,227 @@
+            .appendYear(4, 4)
+            .appendLiteral('-')
+            .appendMonthOfYear(2)
+            .appendLiteral('-')
+            .appendDayOfMonth(2)
+            .appendLiteral(' ')
+            .appendHourOfDay(2)
+            .appendLiteral(':')
+            .appendMinuteOfHour(2)
+            .appendLiteral(':')
+            .appendSecondOfMinute(2)
+            .toFormatter()
+            .withZoneUTC();
+    BIGQUERY_TIMESTAMP_PRINTER =
+        new DateTimeFormatterBuilder()
+            .append(dateTimePart)
+            .appendLiteral('.')
+            .appendFractionOfSecond(3, 3)
+            .appendLiteral(" UTC")
+            .toFormatter();
+  }
+
+  private static Object fromBeamField(FieldType fieldType, Object fieldValue) {
+    if (fieldValue == null) {
+      if (!fieldType.getNullable()) {
+        throw new IllegalArgumentException("Field is not nullable.");
+      }
+      return null;
+    }
+    switch (fieldType.getTypeName()) {
+      case ARRAY:
+      case ITERABLE:
+        FieldType elementType = fieldType.getCollectionElementType();
+        Iterable<?> items = (Iterable<?>) fieldValue;
+        List<Object> convertedItems = Lists.newArrayListWithCapacity(Iterables.size(items));
+        for (Object item : items) {
+          convertedItems.add(fromBeamField(elementType, item));
+        }
+        return convertedItems;
+      case ROW:
+        return toTableRow((Row) fieldValue);
+      case DATETIME:
+        return ((Instant) fieldValue)
+            .toDateTime(DateTimeZone.UTC)
+            .toString(BIGQUERY_TIMESTAMP_PRINTER);
+      case INT16:
+      case INT32:
+      case INT64:
+      case FLOAT:
+      case DOUBLE:
+      case STRING:
+      case BOOLEAN:
+        return fieldValue.toString();
+      case DECIMAL:
+        return fieldValue.toString();
+      case BYTES:
+        return BaseEncoding.base64().encode((byte[]) fieldValue);
+      default:
+        return fieldValue;
+    }
+  }
+
+  public static TableRow toTableRow(Row row) {
+    TableRow output = new TableRow();
+    for (int i = 0; i < row.getFieldCount(); i++) {
+      Object value = row.getValue(i);
+      Field schemaField = row.getSchema().getField(i);
+      output = output.set(schemaField.getName(), fromBeamField(schemaField.getType(), value));
+    }
+    return output;
+  }
+
+  public static TableRow createBqRow(Table.Row tokenizedValue, String[] headers) {
+    TableRow bqRow = new TableRow();
+    AtomicInteger headerIndex = new AtomicInteger(0);
+    // List<TableCell> cells = new ArrayList<>();
+    tokenizedValue
+        .getValuesList()
+        .forEach(
+            value -> {
+              String checkedHeaderName =
+                  Util.checkHeaderName(headers[headerIndex.getAndIncrement()].toString());
+              bqRow.set(checkedHeaderName, value.getStringValue());
+              // cells.add(new TableCell().set(checkedHeaderName, value.getStringValue()));
```

Comment:
so there are several commented out lines of dead code.  did the logic change here? it's like the cells aren't being populated.

Reply:
Hey Elo, we are still working in this PR.

When adding the usage of `BigQueryIO.Write.Method.FILE_LOADS` we started to face errors, so we are testing and investigating  why.